curp/src/members.rs

```diff
@@ -285,14 +285,20 @@ impl ClusterInfo {
         self.cluster_version.load(Ordering::Relaxed)
     }
 
-    /// cluster version increase
-    pub(crate) fn cluster_version_inc(&self) -> u64 {
-        self.cluster_version.fetch_add(1, Ordering::Relaxed)
-    }
-
     /// cluster version decrease
-    pub(crate) fn cluster_version_dec(&self) -> u64 {
-        self.cluster_version.fetch_sub(1, Ordering::Relaxed)
+    pub(crate) fn cluster_version_update(&self) {
+        let mut hasher = std::collections::hash_map::DefaultHasher::new();
+        self.all_members_addrs()
+            .into_iter()
+            .sorted()
+            .for_each(|(id, mut addrs)| {
+                id.hash(&mut hasher);
+                addrs.sort();
+                addrs.hash(&mut hasher);
+            });
+        let ver = hasher.finish();
```

Comment:
I suggest to use at least 128bit hash to avoid collision. FYI: https://stackoverflow.com/questions/62664761/probability-of-hash-collision

Reply:
Protobuf does not like u128. This is somewhat similar to the inflight id.

Reply:
> The birthday paradox specifically says that once you've seen roughly √(2k) items, there's a 50% chance of a collision, where k is the number of distinct possible outputs.
> In the case where the hash function hashes to an n-bit output, this means that you'll need roughly 2^(n/2) hashes before you get a collision.

In our case, 2^(n/2) = 2 ^ 32 = 4,294,967,296, which is a huge number. I think nobody could send configuration changes at this scale.